typing/typecore.ml

```diff
@@ -2006,13 +2053,14 @@ and type_pat_aux
                             loc, sp.ppat_attributes) :: p.pat_extra }
       )
   | Ppat_exception p ->
-      type_pat Value p Predef.type_exn (fun p_exn ->
+      let alloc_mode = simple_pat_mode Value_mode.global in
+      type_pat Value ~alloc_mode p Predef.type_exn (fun p_exn ->
```

Comment:
This looks like an unrelated but good change that improves the typing of match-with-exception - it should probably have an associated test.